### PR TITLE
Add option to export records in benchmark

### PR DESCRIPTION
**Pull request by cbrnr**
By default, the config option `export_records` is `False` so records only exported if explicitly requested.

### CODE REVIEW

**Comment by cbrnr** [comment]
I checked if `runtime` and `metrics` benchmarks yield identical results when loading the records from the original data files versus from exported text files. It seems like this is the case. Runtimes vary a little of course, but no major changes. Metrics look identical (I just compared the plots). I still have the CSV and SVG result files if you want to take a closer look. Other than that, this is ready for review @hofaflo.

**Comment by hofaflo** [comment]
Yes if you could upload the csv results that would be great!

This exports only the raw ecg data, right? Should we export the annotations as well?

**Comment by cbrnr** [comment]
> Yes if you could upload the csv results that would be great!

I'll send them via email.

> This exports only the raw ecg data, right? Should we export the annotations as well?

Good point. Probably yes, but then we need to create multiple files per record. Is this OK?

**Comment by hofaflo** [comment]
> I'll send them via email.

Looks perfect 🎉

> Good point. Probably yes, but then we need to create multiple files per record. Is this OK?

I guess yes 🤔 Alternatively we could use a csv where the first column is the ecg and the second one a `0` or `1` to indicate R peaks. Of course this wouldn't be the most memory efficient solution, but that's a general issue with the plaintext approach :D We could also include the sampling frequency in some kind of header. 

E.g. this code:
```python
import numpy as np
from sleepecg import read_mitdb

rec = next(read_mitdb())
rpeaks = np.zeros_like(rec.ecg)
rpeaks[rec.annotation] = 1

np.savetxt(
    'test2.csv',
    np.vstack((rec.ecg, rpeaks)).T,
    fmt='%.3f,%.0f',
    header=f'# fs: {rec.fs}Hz\necg,rpeak',
    comments='',
)
```

would produce this:

```
# fs: 360Hz 
ecg,rpeak
-0.145,0
-0.145,0
-0.145,0
-0.145,0
-0.145,0
-0.145,0
-0.145,0
-0.145,0
-0.120,0
-0.135,0
...
```

**Comment by hofaflo** [comment]
Great! We could implement this as a function in io/ecg_readers.py (or as a method of ECGReader, but that would break the functional approach we have so far)?

**Comment by cbrnr** [comment]
I guess we could do both? The implementation should be in a function, and we could optionally add a method which just calls that function.

**Comment by cbrnr** [comment]
Something like that? Note that I used a string for the type annotation because we are going to add a method to `ECGRecord`, so this function needs to come before the class – or does it?

**Comment by cbrnr** [comment]
Or maybe we'll skip the method for now, `ECGRecord` is currently a dataclass so we'd also need to change that. I think the function is sufficient for now.

File: examples/benchmark/benchmark_detectors.py [hunk 1]
```diff
@@ -38,12 +42,17 @@
...
 
 data_dir = Path(cfg.get('data_dir', '~/.sleepecg/datasets')).expanduser()
 records = list(reader_dispatch(db_slug, data_dir))
 print(f'Loaded {len(records)} records from {db_slug}.')
 
+if cfg.get('export_records', False):
+    from sleepecg import export_record
+    print(f'Exporting records to {outfile_dir.resolve()}.')
+    for record in records:
+        export_record(record, outfile_dir / f'{record.id}-{record.lead}.txt')
```
Comment (hofaflo) [inline examples/benchmark/benchmark_detectors.py:55]:
```suggestion
        export_record(record, outfile_dir / f'{record.id}-{record.lead}')
```

`export_record` appends `.csv`, so the file would be called e.g. `100-II.txt.csv`

**Comment by hofaflo** [comment]
> Something like that? 

👍 

> Note that I used a string for the type annotation because we are going to add a method to `ECGRecord`, so this function needs to come before the class – or does it?

Functions can be defined below the places they're used, as long as the calling code isn't executed, so both ways would work.

> ECGRecord is currently a dataclass so we'd also need to change that.

Dataclasses can actually have methods, the decorator just creates `__init__`, `__repr__` and comparison dunder methods automatically.

> I think the function is sufficient for now.

I agree, also having only one way is likely less confusing.

Two more notes:
- `export_ecg_record` might be a better name as it's not indended for SleepRecords
- the function should be listed in the docs (i.e. in `api.rst`), either just added to 'Datasets' or we split this category into 'Sleep data' and 'ECG data', WDYT?

File: examples/benchmark/benchmark_detectors.py (hunk 1 above)
Comment (cbrnr) [reply examples/benchmark/benchmark_detectors.py:55]: We could also be a bit more strict, i.e. throw an error if a file has an extension other than `.csv`. We could append `.csv` only if there is no extension.

Comment (hofaflo) [reply examples/benchmark/benchmark_detectors.py:55]: I like that!

**Comment by cbrnr** [comment]
> `export_ecg_record` might be a better name as it's not indended for SleepRecords

Done.

> the function should be listed in the docs (i.e. in `api.rst`), either just added to 'Datasets' or we split this category into 'Sleep data' and 'ECG data', WDYT?

I added it to 'Datasets' for now, splitting would require splitting the menu bar entries in the header as well I assume?

**Comment by hofaflo** [comment]
> I added it to 'Datasets' for now

Seems like that wasn't part of the commit yet :)

> splitting would require splitting the menu bar entries in the header as well I assume?

True, that would be good for consistency, so let's leave it as is.

**Comment by cbrnr** [comment]
I forgot to push the commit. All green now, feel free to merge or let me know if there's anything else.

**Comment by cbrnr** [comment]
Not that it's important, but can/should we link functions in our changelog to the API ref on the docs website? It might be useful unless we have to manually include the corresponding link, in which case it is probably too much effort.

**Comment by hofaflo** [comment]
That would be great, I can't think of a way to have automatic links though :/

**Comment by cbrnr** [comment]
It is possible with .rst but probably not with .md, especially if we want the .md to stay readable e.g. on the GitHub website.

**Comment by hofaflo** [comment]
Hmm... if we keep the chagelog on GitHub I'd prefer to stick with .md, ut if make it part of the docs I think .rst would be fine.

**Comment by hofaflo** [comment]
Anyway this PR is complete, thanks @cbrnr!

**Pull request merged**